docs/file-upload-with-graphql.md

```diff
@@ -0,0 +1,200 @@
+# File upload with graphql
+
+## Overview
+
+This document discuss the issue of uploading files with graphql. \
+As vanilla graphql does not support uploading files, there are several workarounds
+
+### File uploading with gqlgen 0.9.0 library
+<details>
+<summary>Example of schema.graphql</summary>
+<p>
+
+```graphql
+#"The `Upload` scalar type represents a multipart file upload."
+scalar Upload
+
+#"The `File` type, represents the response of uploading a file."
```

Comment:
this example is nice because it presents many possibilities, but maybe we should start with something even simpler:
```
scalar Upload

type Documentation {
    id: ID!
}

type Mutation {
    storeDocumentation(file: Upload!): String
}

type Query {
    anything: String
}
```

and resolver's code:
```
func (r *MyResolver) StoreDocumentation(ctx context.Context, in graphql.Upload) (*string, error) {
	b := bytes.NewBuffer(nil)
	io.Copy(b, in.File)
	out := b.String()
	return &out, nil
}

```